### PR TITLE
pyupgrade + black+ codespell code for the entire repo

### PR DESCRIPTION
**Pull request by jimman2003**
Progress towards #4776 and #5843
~___Blocked___ by #5777 and #5859~
Blocked by #5864
Closes: #5836
Closes: #5835

% `pyupgrade --py39-plus **/*.py`
% `black --version ` # --> black, version 21.10b0 (or later)
% `black --skip-string-normalization .`
%  codespell

### Stakeholders

@cclauss

### CODE REVIEW

**Comment by cclauss** [comment]
Failing tests are like #5853

**Comment by cdrini** [comment]
Can we not merge until #5777 lands ? The rebase will be a nightmare :P Should be merged tomorrow :+1:

**Comment by cdrini** [comment]
After that's merged, you'll have to regenerate the `SolrDocument` : `./openlibrary/solr/types_generator.py > ./openlibrary/solr/solr_types.py` .

**Comment by cclauss** [comment]
Converted to DRAFT until #5777 lands.

**Comment by cdrini** [comment]
#5777 has landed :+1:

**Comment by jimman2003** [comment]
rebase,add solr-types done

**Comment by cclauss** [comment]
Converted to DRAFT again because tests are still failing.  I cannot understand the failure.

**Comment by RayBB** [comment]
This is a weird one for sure. Is anyone able to reproduce it locally? I tried without luck. 

May be worth separating pyupgrade, black,  and codespell into different PRs to isolate the issue.

**Comment by jimman2003** [comment]
Actually I included cclauss's pr about solr types , as I had the error it was trying to solve..

File: openlibrary/solr/update_work.py
Comment (cclauss) [inline openlibrary/solr/update_work.py:1]:
```suggestion
from __future__ import annotations

import datetime
```

File: openlibrary/solr/update_work.py
Comment (cclauss) [inline openlibrary/solr/update_work.py:1383]:
```suggestion
        params={
            'wt': 'json',
            'json.nl': 'arrarr',
            'q': 'author_key:%s' % author_id,
            'sort': 'edition_count desc',
            'row': 1,
            'fl': 'title,subtitle',
            'facet': 'true',
            'facet.mincount': 1,
        }
```

**Review by cclauss — CHANGES_REQUESTED** [review]
requests params are a dict of key / value pairs.

**Review by cclauss — APPROVED** [review]
LGTM.  Thanks much!!

**Pull request merged**
